streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -94,11 +96,11 @@ private JoinWindows(final long beforeMs,
      *
      * @param timeDifference join window interval
      * @throws IllegalArgumentException if {@code timeDifference} is negative or can't be represented as {@code long milliseconds}
+     * @deprecated since 3.0 Use {@link #ofTimeDifferenceWithNoGrace(Duration)} instead
      */
+    @Deprecated
     public static JoinWindows of(final Duration timeDifference) throws IllegalArgumentException {
-        final String msgPrefix = prepareMillisCheckFailMsgPrefix(timeDifference, "timeDifference");
-        final long timeDifferenceMs = validateMillisecondDuration(timeDifference, msgPrefix);
-        return new JoinWindows(timeDifferenceMs, timeDifferenceMs, DEFAULT_GRACE_PERIOD_MS);
+        return ofTimeDifferenceWithNoGrace(timeDifference);
```

Comment:
Have a quick look, the implementation in `ofTimeDifferenceWithNoGrace` is pretty much the same as the old one, but, could you confirm they are ALL the same? I think we should not change deprecated method implementation, since some users are still using them. 

Same questions be below deprecated methods. Thanks.